### PR TITLE
cmake-lint: init at 1.4.3

### LINKED ISSUE #315133: Package request: cmake-lint
**Project description**

Fork of https://github.com/richq/cmake-lint to continue maintenance. cmakelint parses CMake files and reports style issues.

**Metadata**

* homepage URL: https://github.com/cmake-lint/cmake-lint
* source URL: https://github.com/cmake-lint/cmake-lint
* license: apache 2
* platforms: vim plugin

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by luftmensch-luftmensch**
## Description of changes
Closes #315133

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by luftmensch-luftmensch** [comment]
```
> nix build .#cmake-lint
> tree result/
 result
├──  bin
│  ├──  .cmakelint-wrapped
│  └──  cmakelint
├──  lib
│  └──  python3.11
│     └──  site-packages
│        ├──  cmakelint
│        │  ├──  __pycache__
│        │  │  ├──  __init__.cpython-311.opt-1.pyc
│        │  │  ├──  __init__.cpython-311.pyc
│        │  │  ├──  __version__.cpython-311.opt-1.pyc
│        │  │  ├──  __version__.cpython-311.pyc
│        │  │  ├──  main.cpython-311.opt-1.pyc
│        │  │  └──  main.cpython-311.pyc
│        │  ├──  __init__.py
│        │  ├──  __version__.py
│        │  └──  main.py
│        └──  cmakelint-1.4.3.dist-info
│           ├──  entry_points.txt
│           ├──  LICENSE
│           ├──  METADATA
│           ├──  RECORD
│           ├──  top_level.txt
│           └──  WHEEL
└──  nix-support
   └──  propagated-build-inputs
```

Tested also with some samples provided by the cmake-lint repository under `samples/` directory

**Review by MaxwellDupre — APPROVED** [review]
Looks good.

File: pkgs/by-name/cm/cmake-lint/package.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    rev = "refs/tags/${version}";
+    hash = "sha256-/OuWwerBlJynEibaYo+jkLpHt4x9GZrqMRJNxgrDBlM=";
+  };
+
+  nativeBuildInputs = [ python3Packages.setuptools ];
+
+  pythonImportsCheck = [ "cmakelint" ];
+
+  nativeCheckInputs = with python3Packages; [
+    pytest-cov
```
Comment (SuperSandro2000) [inline pkgs/by-name/cm/cmake-lint/package.nix:26]:
```suggestion
```
we don't care about cov at all. Please patch out the requirement

Comment (luftmensch-luftmensch) [reply pkgs/by-name/cm/cmake-lint/package.nix]: Should be good to go now

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**